client/web/web.go

```diff
@@ -695,43 +690,39 @@ func (s *Server) serveAPIAuth(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	resp.Authorized = session.isAuthorized(s.timeNow()) && !caps.isEmpty()
+
 	switch {
 	case sErr != nil && errors.Is(sErr, errNotUsingTailscale):
-		// Restricted to the readonly view, no auth action to take.
 		s.lc.IncrementCounter(r.Context(), "web_client_viewing_local", 1)
-		resp.AuthNeeded = ""
+		resp.Authorized = false // restricted to the readonly view
```

Comment:
I can't quite put my finger on it, but it seems weird to potentially be setting resp.Authorized above, and then setting it back to false here.  In all of these error cases, would `session.isAuthorized` ever return true?  Or can we set resp.Authorized to true in the specific cases in this switch where it applies?  The logic is just a little bit difficult to follow.

Reply:
Yeah that's true, I could _just_ do it on the branch where the user has a valid session. Will do that